packages/core/src/indexing-store/sqlite/store.ts

```diff
@@ -338,6 +342,196 @@ export class SqliteIndexingStore implements IndexingStore {
+            ]),
+          );
+      }
+
+      if (where) {
+        const whereConditions = buildSqlWhereConditions({
+          where,
+          encodeBigInts: true,
+        });
+        for (const whereCondition of whereConditions) {
+          query = query.where(...whereCondition);
+        }
+      }
+
+      if (take) {
+        const limit = validateTake(take);
+        // Get +1 extra row to determine if has next
+        query = query.limit(limit + 1);
+      }
+
+      let orderDirection = "asc";
+
+      if (orderBy) {
+        // Get reversed order conditions, for finding the -1 row
+        const orderByConditions = !before
+          ? buildSqlOrderByConditions({ orderBy })
+          : buildSqlOrderByConditionsReversed({ orderBy });
+
+        orderDirection = orderByConditions[0][1];
+
+        for (const orderByCondition of orderByConditions) {
+          query = query.orderBy(...orderByCondition);
+        }
+        //query = query.orderBy(`id ${orderDirection}`);
+      }
+
+      const { rows } = await this.db.transaction().execute(async (tx) => {
+        if (!!before || !!after) {
+          const currentRowQuery = this.db
+            .selectFrom(table)
+            .selectAll()
+            .limit(1)
+            .where(
+              "id",
+              "=",
+              Buffer.from(after || before || "", "base64").toString(),
+            );
+          const res = await tx.executeQuery(currentRowQuery);
+
+          const dir = orderDirection === "desc" ? "<" : ">";
+
+          if (orderBy && !!res?.rows?.length) {
+            const orderByKey = Object.keys(orderBy)[0];
+            if (orderByKey !== "id") {
+              const resAny = res as any;
+              const orderByValue = resAny.rows[0][orderByKey];
+
+              // Account for ordering by another column besides ID by adding a secondary where -
+              // "where orderByKey > orderByValue or (orderByKey = orderByValue and id > currentId)"
+              query = query.where((eb) =>
+                eb(orderByKey, dir, resAny.rows[0][orderByKey]).or(
+                  eb.and([
+                    eb(orderByKey, "=", orderByValue),
+                    eb("id", dir, resAny.rows[0].id),
+                  ]),
+                ),
+              );
+              // Account for only id ordering - don't need any secondary ordering
+            } else {
+              const resAny = res as any;
+              query = query.where((eb) =>
+                eb(orderByKey, dir, resAny.rows[0][orderByKey]),
+              );
+            }
+          }
+        }
+
+        const selectQuery = await tx.executeQuery(query);
+        return { rows: selectQuery.rows };
+      });
+
+      let deserializedRows = rows.map((row) =>
+        this.deserializeRow({ tableName, row }),
+      );
+      if (before) {
+        deserializedRows.reverse();
+      }
+
+      const hasAfter = rows.length > (take || 1000);
```

Comment:
If I'm reading correctly, this represents "has either before or after", correct?

Considering how different the logic is, and that they're mutually exclusive, I wonder if it would be more readable to write a branch for `after`, `before`, and neither.